src/main/java/com/ivan/salesapp/services/CategoryService.java

```diff
@@ -54,10 +54,14 @@ public CategoryServiceModel editCategory(String id, CategoryServiceModel categor
     }
 
     @Override
-    public void deleteCategory(String id) {
+    public void deleteCategory(String id, IProductService iProductService) {
         Category category = this.categoryRepository.findById(id)
                 .orElseThrow(() -> new IllegalArgumentException("Incorrect id!"));
 
+        if (iProductService.findAllByCategory(category.getName()).size() != 0){
```

Comment:
Extract this category.getName()).size() != 0 before findAllByCategory and then pass it to the method